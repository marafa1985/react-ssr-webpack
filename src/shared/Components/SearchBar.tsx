import React, { useState } from 'react'
import { SearchPanel } from '../Elements'

interface Props {
    onFilterJob: (filtervalue: string) => void
}

const SearchBar: React.FC<Props> = (props) => {
    const [searchValue, setsearchValue] = useState('')
    const filterJobList = (filtervalue: string) => {
        setsearchValue(filtervalue);
        props.onFilterJob(filtervalue);
    }
    return (
        <SearchPanel>
            <h1> JobList </h1>
            <div>
                <input type="text"
                    value={searchValue}
                    onChange={e => filterJobList(e.target.value)}
                    placeholder="Search jobs" />

            </div>
        </SearchPanel>
    )
}

export default SearchBar
