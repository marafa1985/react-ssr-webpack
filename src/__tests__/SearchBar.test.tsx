import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from '../shared/Components/SearchBar';

describe('With Jest', () => {
    it('SearchBar to match snapShot', () => {
        const filterFun = ()=>{};
        const div = document.createElement('div');
        ReactDOM.render(<SearchBar  onFilterJob={filterFun}/>, div);
        expect(div).toMatchSnapshot();
    })
})
