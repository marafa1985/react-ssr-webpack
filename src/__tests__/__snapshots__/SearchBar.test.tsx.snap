// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest SearchBar to match snapShot 1`] = `
<div>
  <div
    class="sc-bdVaJa dqCUNv"
  >
    <h1>
       JobList 
    </h1>
    <div>
      <input
        placeholder="Search jobs"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
